### PR TITLE
FEAT: Support Python 3.10

### LINKED ISSUE #259: Python 3.10 Support
Python 3.10 officially released on 09-Nov-2021 - support for this should be included / verified.

acc crit: 
- [x] have CI/CT running in py3.10 - and pass
- [x] have tox running in py3.10 - and pass
- [x] if it applies: added the py3.10 trove / classifier "Programming Language :: Python :: 3.10" 

depends on https://github.com/CycloneDX/cyclonedx-python-lib/issues/60

### PR DESCRIPTION
**Pull request by jkowalleck**
closes #259

* enabled py3.10 in the CI

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ cyclonedx-bom = 'cyclonedx_py.client:main'
 cyclonedx-py = 'cyclonedx_py.client:main'
 
 [build-system]
-requires = ["poetry-core>=1.0.0"]
+requires = ["poetry-core>=1.0.7"]
```
Comment (jkowalleck) [inline pyproject.toml:34]: :100: this one was required to get a py3.10 install running

**Comment by jkowalleck** [comment]
@madpah i think everything regarding py3.10 was done.
did i forget something?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -68,17 +73,20 @@ jobs:
...
         uses: Gr1N/setup-poetry@v7
         with:
-          poetry-version: 1.1.8
+          poetry-version: ${{ env.POETRY_VERSION }}
+      - name: get poetry virtualenvs
+        id: poetry-virtualenvs
+        run: echo "::set-output name=path::$(poetry config virtualenvs.path)"
       - uses: actions/cache@v2
         with:
-          path: ~/.cache/pypoetry/virtualenvs
```
Comment (jkowalleck) [inline .github/workflows/ci.yml:80]: this path is not correct for windows setups ...

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -68,17 +73,20 @@ jobs:
...
           architecture: 'x64'
       - name: Install poetry
         # see https://github.com/marketplace/actions/setup-poetry
         uses: Gr1N/setup-poetry@v7
         with:
-          poetry-version: 1.1.8
+          poetry-version: ${{ env.POETRY_VERSION }}
+      - name: get poetry virtualenvs
+        id: poetry-virtualenvs
+        run: echo "::set-output name=path::$(poetry config virtualenvs.path)"
```
Comment (jkowalleck) [inline .github/workflows/ci.yml:85]: this way even windows setups will have poper caching 
see https://github.com/CycloneDX/cyclonedx-python/pull/261/files#r750775813

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -28,6 +28,8 @@ on:
 
 env:
   REPORTS_DIR: CI_reports
+  PYTHON_VERISON_DEFAULT: "3.10"
```
Comment (jkowalleck) [inline .github/workflows/ci.yml:31]: @madpah you might want to take this aproach over to the py-lib
using env vars so the several versions could be maintained easily in the github-action

Comment (madpah) [reply .github/workflows/ci.yml:31]: 👍

**Review by madpah — APPROVED** [review]

**Pull request merged**